### DISCUSSION TITLE
How to validate inputs when I don't know how many of them are there?

### DISCUSSION THREAD

**Discussion by galaczi** [adonisjs/core#3752] [Help] [answered]
I am building a server-rendered app and I have a dynamic form on the frontend (with Alpine) where the user can add new rows of inputs via Javascript. So I cannot tell in advance how many rows will be submitted. The shape of each row is the same. How would I go about validating the inputs in each row?

**Comment by galaczi** [comment, marked as answer]
So if I give the same name to the inputs in each row, Adonis puts the values from the same named input in an array. So I can just loop over each array picking out the values for each row. Is this a common practice?

**Reply by WeyIin** [reply]
That's how I would do - or validate whole array at once if that's the possibility. 

But most simple thing is running through the array and validate the fields as if there would be only one of them.
